[Bug]: CLANG build of CpuDxe on ubuntu fails since #128

### Is there an existing issue for this?

- [X] I have searched existing issues

### Current Behavior

Building CpuDxe using CLANG LLVM (AArch64) now fails since #128 due to an uninitialized variable with the following error:

_Extract from the following pipeline:_ https://gus33000.visualstudio.com/30642688-6c52-4bc8-b621-7834caee2e2d/_apis/build/builds/4171/logs/12

```
2023-06-16T07:00:44.0184061Z INFO - "clang" -MMD -MF /home/vsts/work/1/s/Build/SurfaceDuo1-AARCH64/DEBUG_CLANG38/AARCH64/ArmPkg/Drivers/CpuDxe/CpuDxe/OUTPUT/MemoryAttribute.obj.deps -g -Os -fshort-wchar -fno-builtin -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -include AutoGen.h -fno-common -mlittle-endian -fno-short-enums -fverbose-asm -funsigned-char -ffunction-sections -fdata-sections -Wno-address -fno-asynchronous-unwind-tables -fno-unwind-tables -fno-pic -fno-pie -ffixed-x18 -target aarch64-linux-gnu -mcmodel=small -Wno-parentheses-equality -Wno-tautological-compare -Wno-tautological-constant-out-of-range-compare -Wno-empty-body -Wno-unused-const-variable -Wno-varargs -Wno-unknown-warning-option -Wno-unused-but-set-variable -Wno-unused-const-variable -Wno-unaligned-access  -flto -O1 -Wno-unused-const-variable -Wno-sometimes-uninitialized -Wno-unaligned-access -D CLANG -march=armv8.2-a+crypto+rcpc -DUSE_MEMORY_FOR_SERIAL_OUTPUT=0 -DUSE_SCREEN_FOR_SERIAL_OUTPUT=0 -DSECURE_BOOT_ENABLE=1 -c -o /home/vsts/work/1/s/Build/SurfaceDuo1-AARCH64/DEBUG_CLANG38/AARCH64/ArmPkg/Drivers/CpuDxe/CpuDxe/OUTPUT/./MemoryAttribute.obj -I/home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/AArch64 -I/home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe -I/home/vsts/work/1/s/Build/SurfaceDuo1-AARCH64/DEBUG_CLANG38/AARCH64/ArmPkg/Drivers/CpuDxe/CpuDxe/DEBUG -I/home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg -I/home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Include -I/home/vsts/work/1/s/Common/MU_TIANO/EmbeddedPkg -I/home/vsts/work/1/s/Common/MU_TIANO/EmbeddedPkg/Include -I/home/vsts/work/1/s/MU_BASECORE/MdePkg -I/home/vsts/work/1/s/MU_BASECORE/MdePkg/Include -I/home/vsts/work/1/s/MU_BASECORE/MdePkg/Test/UnitTest/Include -I/home/vsts/work/1/s/MU_BASECORE/MdePkg/Test/Mock/Include -I/home/vsts/work/1/s/MU_BASECORE/MdePkg/Include/AArch64 -I/home/vsts/work/1/s/MU_BASECORE/MdeModulePkg -I/home/vsts/work/1/s/MU_BASECORE/MdeModulePkg/Include /home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c
2023-06-16T07:00:44.0854573Z INFO - /home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c:91:15: error: variable 'RegionLength' is uninitialized when used here [-Werror,-Wuninitialized]
2023-06-16T07:00:44.0860866Z INFO -       (UINT64)RegionLength,
2023-06-16T07:00:44.0866415Z INFO -               ^~~~~~~~~~~~
2023-06-16T07:00:44.0872839Z INFO - /home/vsts/work/1/s/MU_BASECORE/MdePkg/Include/Library/DebugLib.h:427:17: note: expanded from macro 'DEBUG'
2023-06-16T07:00:44.0878426Z INFO -         _DEBUG (Expression);       \
2023-06-16T07:00:44.0883977Z INFO -                 ^~~~~~~~~~
2023-06-16T07:00:44.0890080Z INFO - /home/vsts/work/1/s/MU_BASECORE/MdePkg/Include/Library/DebugLib.h:380:42: note: expanded from macro '_DEBUG'
2023-06-16T07:00:44.0895469Z INFO - #define _DEBUG(Expression)  _DEBUG_PRINT Expression
2023-06-16T07:00:44.0899490Z INFO -                                          ^~~~~~~~~~
2023-06-16T07:00:44.0905405Z INFO - /home/vsts/work/1/s/MU_BASECORE/MdePkg/Include/Library/DebugLib.h:377:35: note: expanded from macro '_DEBUG_PRINT'
2023-06-16T07:00:44.0910651Z INFO -         DebugPrint (PrintLevel, ##__VA_ARGS__);      \
2023-06-16T07:00:44.0915587Z INFO -                                   ^~~~~~~~~~~
2023-06-16T07:00:44.0925004Z INFO - /home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c:80:27: note: initialize the variable 'RegionLength' to silence this warning
2023-06-16T07:00:44.0931485Z INFO -   UINTN       RegionLength;
2023-06-16T07:00:44.0935667Z INFO -                           ^
2023-06-16T07:00:44.0940778Z INFO -                            = 0
2023-06-16T07:00:44.0962264Z INFO - 1 error generated.
2023-06-16T07:00:44.0999427Z INFO - make: *** [GNUmakefile:363: /home/vsts/work/1/s/Build/SurfaceDuo1-AARCH64/DEBUG_CLANG38/AARCH64/ArmPkg/Drivers/CpuDxe/CpuDxe/OUTPUT/MemoryAttribute.obj] Error 1
2023-06-16T07:00:44.1011942Z INFO - 
2023-06-16T07:00:44.1012220Z INFO - 
2023-06-16T07:00:44.1012489Z INFO - build.py...
2023-06-16T07:00:44.1012754Z INFO -  : error 7000: Failed to execute command
2023-06-16T07:00:44.1013146Z INFO - 	m a k e   t b u i l d [/home/vsts/work/1/s/Build/SurfaceDuo1-AARCH64/DEBUG_CLANG38/AARCH64/ArmPkg/Drivers/CpuDxe/CpuDxe]
2023-06-16T07:00:46.5301280Z INFO - 
2023-06-16T07:00:46.5323875Z INFO - 
2023-06-16T07:00:46.5324591Z INFO - build.py...
2023-06-16T07:00:46.5325391Z INFO -  : error F002: Failed to build module
2023-06-16T07:00:46.5325979Z INFO - 	/home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/CpuDxe.inf [AARCH64, CLANG38, DEBUG]
2023-06-16T07:00:46.5326771Z INFO - 
2023-06-16T07:00:46.5327411Z INFO - - Failed -
2023-06-16T07:00:46.5327716Z INFO - Build end time: 07:00:46, Jun.16 2023
2023-06-16T07:00:46.5327979Z INFO - Build total time: 00:01:44
2023-06-16T07:00:46.5328176Z INFO - 
2023-06-16T07:00:46.7156084Z INFO - ------------------------------------------------
2023-06-16T07:00:46.7156911Z INFO - --------------Cmd Output Finished---------------
2023-06-16T07:00:46.7160705Z INFO - --------- Running Time (mm:ss): 01:44 ----------
2023-06-16T07:00:46.7169475Z INFO - ----------- Return Code: 0x00000001 ------------
2023-06-16T07:00:46.7169778Z INFO - ------------------------------------------------
2023-06-16T07:00:46.7403662Z ERROR - Compiler #error from /home/vsts/work/1/s/Silicon/Arm/MU_TIANO/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c variable 'RegionLength' is uninitialized when used here [-Werror,-Wuninitialized]
2023-06-16T07:00:46.7404671Z ERROR - Compiler #7000 from : Failed to execute command
2023-06-16T07:00:46.7406289Z ERROR - EDK2 #002 from : Failed to build module
2023-06-16T07:00:46.7408770Z CRITICAL - Build failed
2023-06-16T07:00:46.7410707Z PROGRESS - End time: 2023-06-16 07:00:46.740900	 Total time Elapsed: 0:01:47
```

### Expected Behavior

The build should not be failing using CLANG LLVM targeting AArch64.

### Steps To Reproduce

1. Use the following environment: Ubuntu latest (AzDo hosted image)
2. Install clang dependencies using ```sudo apt install -y uuid-dev clang llvm gcc-aarch64-linux-gnu```
3. Export CLANG variable using ```export CLANG38_BIN=/usr/lib/llvm-38/bin/ && export CLANG38_AARCH64_PREFIX=aarch64-linux-gnu-```
4. Try to build a project making use of CpuDxe with a debug level compiling the new logs added in #128 
5. Observe the building like above

### Build Environment

```markdown
- OS(s): Ubuntu 22.04.2 LTS
- Tool Chain(s): CLANG AARCH64 LINUX GNU
- Targets Impacted: DEBUG
```


### Version Information

```text
Commit: f4665e122970076929ca4c9513b120dc7b0ced69
```


### Urgency

Medium

### Are you going to fix this?

Someone else needs to fix it

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

The pipeline can be found here: https://gus33000.visualstudio.com/SurfaceDuoPkg/_build/results?buildId=4171&view=results

I have not migrated to CLANGPDB from CLAN38 (in terms of naming for the build system) but am using the latest clang version available for Ubuntu 22.04.2 LTS from APT sources. I do not know however if this could be causing a problem.

--- a/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c
+++ b/ArmPkg/Drivers/CpuDxe/MemoryAttribute.c
@@ -84,13 +84,13 @@ GetMemoryAttributes (
   EFI_STATUS  Status;
 
   if ((Length == 0) || (Attributes == NULL)) {
+    // MU_CHANGE START: Update protocol logging
     DEBUG ((
       DEBUG_ERROR,
-      "%a invalid length 0x%llx or attributes %p are null\n",
-      __func__,
-      (UINT64)RegionLength,
-      Attributes
-      )); // MU_CHANGE: Better memory attributes protocol logging
+      "%a: Length is zero or Attributes is NULL\n",
+      __func__
+      ));
+    // MU_CHANGE END
     return EFI_INVALID_PARAMETER;
   }
 
@@ -208,13 +208,14 @@ SetMemoryAttributes (
   if ((Length == 0) ||
       ((Attributes & ~(EFI_MEMORY_RO | EFI_MEMORY_RP | EFI_MEMORY_XP)) != 0))
   {
+    // MU_CHANGE START: Update protocol logging
     DEBUG ((
       DEBUG_ERROR,
-      "%a invalid length: 0x%llx or attributes 0x%llx\n",
+      "%a: Length is zero or Attributes (0x%Lx) is invalid\n",
       __func__,
-      Length,
       Attributes
-      )); // MU_CHANGE: Better memory attributes protocol logging
+      ));
+    // MU_CHANGE END
     return EFI_INVALID_PARAMETER;
   }
 
@@ -227,14 +228,16 @@ SetMemoryAttributes (
   if ((Attributes & EFI_MEMORY_RP) != 0) {
     Status = ArmSetMemoryRegionNoAccess (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to set EFI_MEMORY_RP on BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to set EFI_MEMORY_RP on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }
@@ -242,14 +245,16 @@ SetMemoryAttributes (
   if ((Attributes & EFI_MEMORY_RO) != 0) {
     Status = ArmSetMemoryRegionReadOnly (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to set EFI_MEMORY_RO on BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to set EFI_MEMORY_RO on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }
@@ -257,14 +262,16 @@ SetMemoryAttributes (
   if ((Attributes & EFI_MEMORY_XP) != 0) {
     Status = ArmSetMemoryRegionNoExec (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to set EFI_MEMORY_XP on BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to set EFI_MEMORY_XP on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }
@@ -325,13 +332,14 @@ ClearMemoryAttributes (
   if ((Length == 0) ||
       ((Attributes & ~(EFI_MEMORY_RO | EFI_MEMORY_RP | EFI_MEMORY_XP)) != 0))
   {
+    // MU_CHANGE START: Update protocol logging
     DEBUG ((
       DEBUG_ERROR,
-      "%a invalid length: 0x%llx or attributes 0x%llx\n",
+      "%a: Length is zero or Attributes (0x%Lx) is invalid\n",
       __func__,
-      Length,
       Attributes
-      )); // MU_CHANGE: Better memory attributes protocol logging
+      ));
+    // MU_CHANGE END
     return EFI_INVALID_PARAMETER;
   }
 
@@ -344,14 +352,16 @@ ClearMemoryAttributes (
   if ((Attributes & EFI_MEMORY_RP) != 0) {
     Status = ArmClearMemoryRegionNoAccess (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to clear EFI_MEMORY_RP from BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to clear EFI_MEMORY_RP on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }
@@ -359,14 +369,16 @@ ClearMemoryAttributes (
   if ((Attributes & EFI_MEMORY_RO) != 0) {
     Status = ArmClearMemoryRegionReadOnly (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to clear EFI_MEMORY_RO from BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to clear EFI_MEMORY_RO on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }
@@ -374,14 +386,16 @@ ClearMemoryAttributes (
   if ((Attributes & EFI_MEMORY_XP) != 0) {
     Status = ArmClearMemoryRegionNoExec (BaseAddress, Length);
     if (EFI_ERROR (Status)) {
+      // MU_CHANGE START: Update protocol logging
       DEBUG ((
         DEBUG_ERROR,
-        "%a failed to clear EFI_MEMORY_XP from BaseAddress 0x%llx of length 0x%llx with status %r!\n",
+        "%a: Failed to clear EFI_MEMORY_XP on range 0x%Lx-0x%Lx! Status: %r\n",
         __func__,
         BaseAddress,
-        Length,
+        BaseAddress + Length,
         Status
-        )); // MU_CHANGE: Better memory attribute protocol logging
+        ));
+      // MU_CHANGE END
       return EFI_UNSUPPORTED;
     }
   }

